rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+AccordionHeaderSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  button: ARIAButtonShorthandProps;
+  expandIcon: AccordionHeaderExpandIconProps;
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  children: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordianItemSlots, AccordianPanelSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuSlots
+
+```ts
+(empty type)
+```
+
+MenuDividerSlots, MenuGroupSlots, MenuGroupHeaderSlots, MenuListSlots, MenuPopoverSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuItemSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+icon?: React.HTMLAttributes<HTMLElement>;
+checkmark?: React.HTMLAttributes<HTMLElement>;
+submenuIndicator?: React.HTMLAttributes<HTMLElement>;
+content: React.HTMLAttributes<HTMLElement>;
+secondaryContent?: React.HTMLAttributes<HTMLElement>;
+```
+
+## Compat and HTML attributes
+
+> The components not yet using slots extend ComponentPropsCompat and provide HTML attributes by extending the associated React HTML attribute types (e.g `React.HTMLAttributes<HTMLElement>`).
+> PortalProps and PopoverTriggerProps do not extend any HTML attributes, but provide children as a property directly in their types.
+
+AvatarProps,BadgeProps,DividerProps, TooltipProps
+
+```ts
+extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement>
+```
+
+ButtonProps
+
+```ts
+ButtonProps = ComponentPropsCompat & React.ButtonHTMLAttributes<HTMLElement> & {...}
+```
+
+MenuButtonProps/MenuTriggerChildProps
+
+```ts
+MenuButtonProps = Omit<ButtonProps, 'iconPosition'>  & Partial<MenuTriggerChildProps> & {}
+
+MenuTriggerChildProps extends Required<Pick<React.HTMLAttributes<HTMLElement>,
+      | 'onClick'
+      | 'onMouseEnter'
+      | 'onMouseLeave'
+      | 'onContextMenu'
+      | 'onKeyDown'
+      | 'aria-haspopup'
+      | 'aria-expanded'
+      | 'id'
+    >
+```
+
+ImageProps
+
+```ts
+extends ComponentPropsCompat,
+React.ImgHTMLAttributes<HTMLImageElement>
+```
+
+LabelProps
+
+```ts
+extends ComponentPropsCompat,
+React.LabelHTMLAttributes<HTMLLabelElement>
+```
+
+LinkProps
+
+```ts
+LinkProps = ComponentPropsCompat &
+  React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement> &
+  Omit<React.ButtonHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement>, 'type'> & {...}
+```
+
+PopoverProps
+
+```ts
+// PortalProps is an interface with children and mountNode.
+// It does not provide HTML attributes
+extends Pick<PortalProps, 'mountNode'>
+```
+
+PopoverSurfaceProps
+
+```ts
+extends ComponentPropsCompat,
+React.HTMLAttributes<HTMLElement>
+```
+
+## Property overrides of native properties
+
+> Several components declare properties that are also declared by extending from the DOM types (e.g. React.HtmlAttributes<T>).
```

Comment:
from @ecraig12345 in HackMD:
This is an interesting one, and a very good case for conformance testing. There are a couple sub-cases:
1. Explicitly declaring (as original type) for documentation purposes => probably fine (example: people get confused if the docs for Link don't explicitly mention `href`)
2. Explicitly declaring as limited sub-type => TBD if this is okay. If we allow it, it should be paired with Omit on the native props.
3. Accidental reuse of name for a different purpose => should come up with a different name (example: for Input I wanted to use `size` but had to rename it since that's a native input prop with different meaning)